### DISCUSSION TITLE
Question about information `--stats-internal`

### DISCUSSION THREAD

**Discussion by hagozaebii** [cvc5/cvc5#11265] [Q&A] [answered]
Hello, Recently I had an interest in statistics about formulae. I tried to get stats info with `--stats-internal` for 1 second. The results are as follows.
```
sat::clauses_literals = 8507
sat::conflicts = 12
sat::decisions = 2860
sat::learnts_literals = 336
sat::max_literals = 147
sat::propagations = 38789
sat::starts = 1
sat::tot_literals = 116
```
I want to know about this information. What is the meaning of `tot_literals` and `max_literals`? `tot_literals` means `the number of current literals in the termination point`? If so, does `max_literals` mean `the maximal number of literals while solving`?

Thanks for your time!

**Comment by mpreiner** [comment, marked as answer]
I had a quick look at the code since the documentation is missing for these statistics. I think that these  two statistics refer to the number of literals for learnt clauses before clause minimization (max_literals) and number of literals after (tot_literals). The code for this can be found here: https://github.com/cvc5/cvc5/blob/main/src/prop/minisat/core/Solver.cc#L1003

**Comment by hagozaebii** [comment]
@mpreiner First of all, thanks for your kindness. It helps a lot :)

However, I still have a question. if you are okay, could you let me know something more?
I guess that `clause minimization` is not executed just once while solving. If my thought is right, then `max literals` means the number of literals before `last minimization`? And `total literals` are the number of clause after `the last minimization`?
